lib/distributed_cache.rb

```diff
@@ -14,6 +14,8 @@ def defer_set(k, v)
   end
 
   def defer_get_set(k, &block)
+    raise TypeError unless Rails.env.production? || String === k
```

Comment:
`unless` mixed with `||`?  😭 (my kryptonite! 😂) how about:

```suggestion
    raise TypeError if !Rails.env.production? && !k.is_a?(String)
```

Reply:
De Morgan approves :smiling_face_with_tear: